napari/layers/points/points.py

```diff
@@ -1269,7 +1288,11 @@ def _get_value(self, position) -> Union[None, int]:
         if len(view_data) > 0:
             displayed_position = [position[i] for i in self._dims_displayed]
             # Get the point sizes
-            distances = abs(view_data - displayed_position)
+            if not self.fixed_canvas_size:
+                distances = abs(view_data - displayed_position)
+            else:
+                # TODO: calculate distance in canvas space
+                return None
```

Comment:
This is where we should calculate distances differently. @alisterburt @kevinyamauchi , I'm not sure where to get the right transforms. I see we have data_to_world and friends, but do we expose any canvas data?
Alternatively, how would you approach this?

Reply:
I think it depends on whether we want to support perspective or not. For orthographic projection, we can use the camera zoom to calculate the canvas pixel size in data units and use that. The tricky part is getting access to the scale layer.

See how the calculation is done for the axis visual: https://github.com/napari/napari/blob/fc133a627dc92aff0297a010d7e2e4f5d3a82092/napari/_vispy/overlays/axes.py#L328-L343

Reply:
But we don't have access to viewer and camera from the layer, and we probably shouldn't... This might be a problem :P

Reply:
Yeah, agreed. This is one of the reasons we have been punting on interactivity with perspective. I would be okay with proceeding without for now seeing as perspective breaks interactivity for all other layers.

Reply:
I can review more carefully tomorrow!

Reply:
To be clear: this is not an issue only with perspective. Even with fov=0, without additional changes and with `fixed_canvas_size` active we get basically the same issue as #3610.